security/pkg/nodeagent/caclient/client.go

```diff
@@ -56,7 +57,7 @@ func NewCAClient(endpoint, caProviderName string, tlsFlag bool, tlsRootCert []by
 		if err != nil {
 			return nil, fmt.Errorf("could not create k8s clientset: %v", err)
 		}
-		controller := configmap.NewController(ns, cs.CoreV1())
+		controller := configmap.NewController(namespace, cs.CoreV1())
```

Comment:
Is everything we are reading from always in the same namespace as the nodeagent? We have the concept of the "rootNamespace" which has all the global istio configs. So for example `*.local` destination rule, default sidecar, etc all live there. Should this be reading from that or the nodeagent namespace?

Reply:
should be nodeagent namesapce (citadel write root cert to configmap where it's deployed, nodeagent read from it, and use it to connect to citadel)